Mathlib/Geometry/Euclidean/Triangle.lean

```diff
@@ -262,6 +287,37 @@ theorem dist_sq_eq_dist_sq_add_dist_sq_sub_two_mul_dist_mul_dist_mul_cos_angle (
 
 alias law_cos := dist_sq_eq_dist_sq_add_dist_sq_sub_two_mul_dist_mul_dist_mul_cos_angle
 
+/-- **Law of sines** (sine rule), angle-at-point form. -/
+theorem sin_angle_mul_dist_eq_sin_angle_mul_dist {p₁ p₂ p₃ : P} :
+    Real.sin (∠ p₁ p₂ p₃) * dist p₂ p₃ = Real.sin (∠ p₃ p₁ p₂) * dist p₃ p₁ := by
+  simp only [dist_comm p₂ p₃, angle]
+  rw [dist_eq_norm_vsub V p₃ p₂, dist_eq_norm_vsub V p₃ p₁]
+  rw [InnerProductGeometry.angle_comm, sin_angle_mul_norm_eq_sin_angle_mul_norm]
+  rw [vsub_sub_vsub_cancel_right, mul_eq_mul_right_iff]
+  left
+  rw [InnerProductGeometry.angle_comm, ← neg_vsub_eq_vsub_rev p₁ p₂]
+  rw [angle_neg_right, Real.sin_pi_sub]
+
+alias law_sin := sin_angle_mul_dist_eq_sin_angle_mul_dist
+
+/-- A variant of the law of sines, angle-at-point form. -/
+theorem sin_angle_div_dist_eq_sin_angle_div_dist {p₁ p₂ p₃ : P} (h23: p₂ ≠ p₃) (h31: p₃ ≠ p₁):
+    Real.sin (∠ p₁ p₂ p₃) / dist p₃ p₁ = Real.sin (∠ p₃ p₁ p₂) / dist p₂ p₃ := by
+  field_simp [dist_ne_zero.mpr h23, dist_ne_zero.mpr h31]
+  exact law_sin
+
+/-- A variant of the law of sines, requiring that the points not be collinear. -/
+theorem dist_eq_dist_mul_sin_angle_div_sin_angle {p₁ p₂ p₃ : P}
+    (h : AffineIndependent ℝ ![p₁,p₂,p₃]) :
```

Comment:
@jsm28, should this be stated with "not collinear" instead?

Reply:
"not collinear" is probably more convenient when you have the hypothesis with the points in a different order and need to reorder them before applying this result. `AffineIndependent` would allow dot notation if this result were put into that namespace (and you have the points in the right order), but I'm not sure that's a particularly significant benefit.